inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/reification/NoReification.java

```diff
@@ -0,0 +1,247 @@
+            catch (QueryEvaluationException e) {
+                log.warn("Listing statements failed.", e);
+                return Collections.emptyList();
+            }
+
+            List<Statement> statements = new ArrayList<>();
+            IRI subject = vf.createIRI(aInstanceIdentifier);
+            while (result.hasNext()) {
+                BindingSet bindings = result.next();
+                Binding pred = bindings.getBinding("p");
+                Binding obj = bindings.getBinding("o");
+
+                IRI predicate = vf.createIRI(pred.getValue().stringValue());
+                Statement stmt = vf.createStatement(subject, predicate, obj.getValue());
+                statements.add(stmt);
+            }
+            return statements;
+        }
+    }
+
+    @Override
+    public void deleteStatement(KnowledgeBase kb, KBStatement aStatement)
+    {
+        update(kb, (conn) -> {
+            conn.remove(aStatement.getOriginalStatements());
+            aStatement.setOriginalStatements(Collections.emptyList());
+            return null;
+        });
+    }
+
+    @Override
+    public void upsertStatement(KnowledgeBase kb, KBStatement aStatement)
+    {
+        update(kb, (conn) -> {
+            if (!aStatement.isInferred()) {
+                conn.remove(aStatement.getOriginalStatements());
+            }
+            List<Statement> statements = reify(kb, aStatement);
+            conn.add(statements);
+            aStatement.setOriginalStatements(statements);
+
+            return null;
+        });
+    }
+
+    @Override
+    public void addQualifier(KnowledgeBase kb, KBQualifier newQualifier)
+    {
+        log.error("Qualifiers are not supported.");
+    }
+
+    @Override
+    public void deleteQualifier(KnowledgeBase kb, KBQualifier oldQualifier)
+    {
+        log.error("Qualifiers are not supported.");
+    }
+
+    @Override
+    public void upsertQualifier(KnowledgeBase kb, KBQualifier aQualifier)
+    {
+        log.error("Qualifiers are not supported.");
+    }
+
+    @Override
+    public List<KBQualifier> listQualifiers(KnowledgeBase kb, KBStatement aStatement)
+    {
+        log.error("Qualifiers are not supported.");
+        return null;
+    }
+
+    private KBHandle update(KnowledgeBase kb, UpdateAction aAction)
+    {
+        if (kb.isReadOnly()) {
+            log.warn("Knowledge base [{}] is read only, will not alter!", kb.getName());
+            return null;
+        }
+
+        KBHandle result = null;
+        try (RepositoryConnection conn = kbService.getConnection(kb)) {
+            boolean error = true;
+            try {
+                conn.begin();
+                result = aAction.accept(conn);
+                conn.commit();
+                error = false;
+            }
+            finally {
+                if (error) {
+                    conn.rollback();
+                }
+            }
+        }
+        return result;
+    }
+
+    private interface UpdateAction
```

Comment:
Kannst du nicht das update vom KBService benutzen?

Reply:
`update` from KBService is private. Could I change it to public?